R/new_ticker.R

```diff
@@ -0,0 +1,33 @@
+new_ticker <- function(label, n, progress = NA) {
+  # if progress bar should'nt be shown, return identity
+  # message or fail depending on `progress`
+
+  if (is.na(progress)) {
+    progress <- interactive()
+    if (progress && !requireNamespace("progress")) {
+      message("The 'progress' package must be installed in order to display progress bars.")
+      return(identity)
+    }
+  }
+
+  if (!progress) return(identity)
+
+  if (!requireNamespace("progress")) {
+    stop("The 'progress' package must be installed in order to display progress bars.")
```

Comment:
Ultimately, this should be `abort()`. I wonder if we should implement

```r
check_suggested <- function(package, use, message) {
  if (!use) {
    return(FALSE)
  } else if (is.na(use)) {
    if (is_installed(package)) {
  ...
}
```

so that this pattern can be applied consistently and we can move more packages to "Suggests" if needed?

Reply:
good idea, it's also more readable this way, I've implemented it

Reply:
Once we merge this we should use `check_suggested()` in `bdm_render_graph()` to replace :

```
  if (!rlang::is_installed("DiagrammeR")) {
    stop("DiagrammeR package needed for this function to work. Please install it.",
      call. = FALSE
    )
  }
```